portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -51,10 +61,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Map<String, Map<String, String>> comments = new HashMap<>();
     for (Entity entity : results.asIterable()) {
       String id = KeyFactory.keyToString(entity.getKey());
-      Map<String, String> commentEmail = new HashMap<>();
-      commentEmail.put("comment", (String) entity.getProperty("comment"));
-      commentEmail.put("userEmail", (String) entity.getProperty("userEmail"));
-      comments.put(id, commentEmail);
+      Map<String, String> commentAttributes = new HashMap<>();
+      commentAttributes.put("comment", (String) entity.getProperty("comment"));
+      commentAttributes.put("userEmail", (String) entity.getProperty("userEmail"));
+      commentAttributes.put("imageUrl", (String) entity.getProperty("imageUrl"));
```

Comment:
If I read the code correctly, we allow client to send comments without any image.
When the client tries to retrieve the comment, they would get a json response with the string 'null' for imageURL.

Am I correct?

Reply:
Yes that is correct

Reply:
That would force the client to try comparing the returned imageURL to this special value, which might be the less intuitive API.

It is ok, but perhaps a better design would not assign/send this attribute with a special value (the client can then check if the returned value has the imageURL attribute using https://www.w3schools.com/jsref/met_element_hasattribute.asp )

Reply:
How would you like to address this?

Reply:
I'm not quite sure I understand what you mean by this. This method applies to the HTML dom elements, this is the map of comments and their corresponding information that I am passing to the JSON function that will eventually be read in the JS file and dynamically create the comments on the page. Could you explain what you would like to see here a little more please?